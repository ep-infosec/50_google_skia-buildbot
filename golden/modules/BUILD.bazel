load("//infra-sk:index.bzl", "karma_test", "sass_library", "ts_library")

karma_test(
    name = "common_test",
    src = "common_test.ts",
    deps = [
        ":common_ts_lib",
        "//golden/modules/search-controls-sk",
        "//infra-sk/modules:test_util_ts_lib",
        "@npm//@types/chai",
        "@npm//chai",
        "@npm//common-sk",
    ],
)

ts_library(
    name = "common_ts_lib",
    srcs = ["common.ts"],
    visibility = ["//visibility:public"],
    deps = [
        ":rpc_types_ts_lib",
        "//golden/modules/search-controls-sk",
        "@npm//common-sk",
    ],
)

ts_library(
    name = "demo_util_ts_lib",
    srcs = ["demo_util.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "rpc_types_ts_lib",
    srcs = ["rpc_types.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "settings_ts_lib",
    srcs = ["settings.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "shared_demo_data_ts_lib",
    srcs = ["shared_demo_data.ts"],
    visibility = ["//visibility:public"],
    deps = ["@npm//common-sk"],
)

sass_library(
    name = "colors_sass_lib",
    srcs = ["colors.scss"],
    visibility = ["//visibility:public"],
)
