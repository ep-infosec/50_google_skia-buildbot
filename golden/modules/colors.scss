/* Common Gold Colors via CSS Variables.

 Can be imported via:

    @import '../colors';
*/

html {
  --gold: rgb(217, 95, 2);
  --bright-orange: rgb(255, 127, 0);

  --light-error-red: rgb(255, 230, 230);
  --transparent-white: rgba(255, 255, 255, 0);

  --triaged-positive: #1B9E77;
  --triaged-negative: #E7298A;
  --untriaged: #A6761D;
}
