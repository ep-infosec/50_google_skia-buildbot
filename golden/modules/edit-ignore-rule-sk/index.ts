import './edit-ignore-rule-sk';
