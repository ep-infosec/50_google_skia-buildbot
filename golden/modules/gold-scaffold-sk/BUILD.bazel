load("//infra-sk:index.bzl", "karma_test", "sk_demo_page_server", "sk_element", "sk_element_puppeteer_test", "sk_page")

sk_demo_page_server(
    name = "demo_page_server",
    sk_page = ":gold-scaffold-sk-demo",
)

sk_element(
    name = "gold-scaffold-sk",
    sass_deps = [
        "//golden/modules:colors_sass_lib",
        "@npm//:node_modules/elements-sk/colors.scss",
    ],
    sass_srcs = ["gold-scaffold-sk.scss"],
    sk_element_deps = [
        "//golden/modules/last-commit-sk",
        "//infra-sk/modules/app-sk",
        "//infra-sk/modules/login-sk",
    ],
    ts_deps = [
        "//golden/modules:common_ts_lib",
        "//golden/modules:settings_ts_lib",
        "//infra-sk/modules/ElementSk:index_ts_lib",
        "@npm//elements-sk",
        "@npm//lit-html",
    ],
    ts_srcs = [
        "gold-scaffold-sk.ts",
        "index.ts",
    ],
    visibility = ["//visibility:public"],
)

sk_page(
    name = "gold-scaffold-sk-demo",
    html_file = "gold-scaffold-sk-demo.html",
    sk_element_deps = [":gold-scaffold-sk"],
    ts_deps = [
        "//golden/modules:settings_ts_lib",
        "@npm//common-sk",
    ],
    ts_entry_point = "gold-scaffold-sk-demo.ts",
)

sk_element_puppeteer_test(
    name = "gold-scaffold-sk_puppeteer_test",
    src = "gold-scaffold-sk_puppeteer_test.ts",
    sk_demo_page_server = ":demo_page_server",
    deps = [
        "//puppeteer-tests:util_ts_lib",
        "@npm//@types/chai",
        "@npm//chai",
    ],
)

karma_test(
    name = "gold-scaffold-sk_test",
    src = "gold-scaffold-sk_test.ts",
    deps = [
        ":gold-scaffold-sk",
        "//golden/modules:common_ts_lib",
        "//golden/modules:settings_ts_lib",
        "//golden/modules/last-commit-sk:demo_data_ts_lib",
        "//infra-sk/modules:test_util_ts_lib",
        "@npm//@types/chai",
        "@npm//chai",
        "@npm//common-sk",
        "@npm//elements-sk",
        "@npm//fetch-mock",
    ],
)
