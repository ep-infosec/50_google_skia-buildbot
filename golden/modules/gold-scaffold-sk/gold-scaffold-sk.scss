@import 'node_modules/elements-sk/colors.scss';
@import '../colors';

gold-scaffold-sk {
  font-family: Roboto, Helvetica, Arial, 'Bitstream Vera Sans', sans-serif;

  app-sk {
    --header-background-color: var(--gold);
    --header-font-color: var(--on-primary, var(--white));
    --background: var(--white);

    main {
      // Prevent the app-sk from stretching horizontally indefinitely (for example, dots-sk can
      // become very long). See https://css-tricks.com/preventing-a-grid-blowout/.
      min-width: 0;
    }
  }

  header {
    h1 {
      font-weight: bold;
      font-size: 1.5em;
    }

    .spinner-spacer {
      min-width: 40px;
    }

    spinner-sk {
      width: 24px;
      height: 24px;
      margin: 0;
      border: 8px solid var(--light-orange);
      border-left: 8px solid var(--red);
    }

    last-commit-sk {
      color: var(--white);
      margin-right: 5px;
    }

    login-sk {
      margin-left: auto;
      margin-right: 1em;
      display: block;
      color: var(--white);
    }
  }

  nav {
    display: flex;
    flex-direction: column;
    a {
      padding: 0.5em 0.75em;
      color: var(--black);
      fill: var(--black);
      text-decoration: none;
    }
    span {
      padding: 0 0.5em;
      vertical-align: middle;
    }
  }
}
