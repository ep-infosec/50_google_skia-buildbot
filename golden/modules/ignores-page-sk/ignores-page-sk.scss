@import 'node_modules/elements-sk/colors.scss';
@import '../colors';

ignores-page-sk {
  .controls {
    display: flex;
    padding: 4px 0;
    *, checkbox-sk .box {
      margin: auto 5px;
    }
  }

  .small-icon > .icon-sk-svg {
    width: 20px;
    height: 20px;
  }

  table {
    border-collapse: collapse;
    min-width: 100%;
    margin-bottom: 10px;
  }

  th {
    border-bottom: 2px solid black;
  }

  tr {
    &.expired {
      background-color: var(--light-error-red);
    }
  }

  td {
    border: 1px solid var(--dark-white);
    border-bottom: 1px solid var(--light-gray);
    text-align: center;
    padding: 6px 2px;

    &.mutate-icons {
      *:hover{
        fill: var(--gold);
        cursor: pointer;
      }
    }
    &.query {
      white-space: pre;
      vertical-align: middle;
      text-align: left;
    }
    &.expired {
      font-weight: bolder;
    }
  }
}
