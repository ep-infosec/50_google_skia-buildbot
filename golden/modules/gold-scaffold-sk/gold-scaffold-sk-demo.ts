import './index';
import { $$ } from 'common-sk/modules/dom';
import { testOnlySetSettings } from '../settings';
import { GoldScaffoldSk } from './gold-scaffold-sk';

testOnlySetSettings({
  title: 'Skia Public',
});

// Remove from DOM and reattach to trigger a re-render. This ensures that the GoldScaffoldSk
// instance will pick up the title from the above settings.
const goldScaffoldSk = $$<GoldScaffoldSk>('gold-scaffold-sk')!;
const parent = goldScaffoldSk.parentNode!;
parent.removeChild(goldScaffoldSk);
parent.appendChild(goldScaffoldSk);
