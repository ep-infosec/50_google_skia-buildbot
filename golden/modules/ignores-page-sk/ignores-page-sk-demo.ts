import './index';
import '../gold-scaffold-sk';

import fetchMock from 'fetch-mock';
import { delay } from '../demo_util';
import { ignoreRules_10, fakeNow } from './test_data';
import { manyParams } from '../shared_demo_data';
import { testOnlySetSettings } from '../settings';
import { exampleStatusData } from '../last-commit-sk/demo_data';
import { GoldScaffoldSk } from '../gold-scaffold-sk/gold-scaffold-sk';

Date.now = () => fakeNow;
testOnlySetSettings({
  title: 'Skia Public',
  baseRepoURL: 'https://skia.googlesource.com/skia.git',
});

fetchMock.get('/json/v2/paramset', delay(manyParams, 100));
fetchMock.get('/json/v2/ignores', delay(ignoreRules_10, 300));
fetchMock.post('glob:/json/v1/ignores/del/*', delay({}, 600));
fetchMock.post('glob:/json/v1/ignores/add/', delay({}, 600));
fetchMock.post('glob:/json/v1/ignores/save/*', delay({}, 600));
fetchMock.get('/json/v2/trstatus', JSON.stringify(exampleStatusData));

// By adding these elements after all the fetches are mocked out, they should load ok.
const newScaf = new GoldScaffoldSk();
newScaf.testingOffline = true;
// Make it the first element in body.
document.body.insertBefore(newScaf, document.body.childNodes[0]);
const page = document.createElement('ignores-page-sk');
page.setAttribute('page_size', '10');
newScaf.appendChild(page);
