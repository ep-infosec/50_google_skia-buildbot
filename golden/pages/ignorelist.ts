import '../modules/ignores-page-sk';
import '../modules/gold-scaffold-sk';
