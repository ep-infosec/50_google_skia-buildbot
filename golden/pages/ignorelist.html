<!DOCTYPE html>
<html>
  <head>
    <title>{{.Title}} Ignore Rules</title>
    <link rel="shortcut icon" href="/dist/favicon.ico" />
    <script>
      window.GoldSettings = {
        title: '{{.Title}}',
        defaultCorpus: '{{.DefaultCorpus}}',
        baseRepoURL: '{{.BaseRepoURL}}',
      };
    </script>
  </head>
  <body class="body-sk">
    <gold-scaffold-sk>
      <ignores-page-sk></ignores-page-sk>
    </gold-scaffold-sk>
  </body>
</html>
